src/convertToObject.js

```diff
@@ -6,7 +6,28 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const cleanedString = sourceString.trim();
```

Comment:
[CODE KNOWLEDGE] - Don't need to make copy before each operation. Moreover, methods like `filter` or `map` don't mutate original array. In this case, `trim` is not mutating `sourceString`, but the variable `cleanedString` is unnecessary since you can call `trim` directly where it's needed.